plugins/woocommerce-admin/client/marketing/overview-multichannel/Campaigns/CreateNewCampaignModal/CreateNewCampaignModal.test.tsx

```diff
@@ -0,0 +1,151 @@
+					direct_install: true,
+					icon: 'https://woocommerce.com/wp-content/plugins/wccom-plugins/marketing-tab-rest-api/icons/pinterest.svg',
+					product: 'pinterest-for-woocommerce',
+					plugin: 'pinterest-for-woocommerce/pinterest-for-woocommerce.php',
+					categories: [ 'marketing' ],
+					subcategories: [
+						{ slug: 'sales-channels', name: 'Sales channels' },
+					],
+					tags: [
+						{
+							slug: 'built-by-woocommerce',
+							name: 'Built by WooCommerce',
+						},
+					],
+					show_extension_promotions: true,
+				},
+				{
+					title: 'Amazon, eBay & Walmart Integration for WooCommerce',
+					description:
+						'Convert Woocommerce into a fully-featured omnichannel commerce platform, leveraging powerful automation and real-time sync to connect your brand with millions of new customers on the world\u2019s largest online marketplaces.',
+					url: 'https://woocommerce.com/products/amazon-ebay-integration/?utm_source=marketingtab&utm_medium=product&utm_campaign=wcaddons',
+					direct_install: false,
+					icon: 'https://woocommerce.com/wp-content/plugins/wccom-plugins/marketing-tab-rest-api/icons/amazon-ebay.svg',
+					product: 'amazon-ebay-integration',
+					plugin: 'woocommerce-amazon-ebay-integration/woocommerce-amazon-ebay-integration.php',
+					categories: [ 'marketing' ],
+					subcategories: [
+						{ slug: 'sales-channels', name: 'Sales channels' },
+					],
+					tags: [],
+				},
+			],
+		} );
+		( useRegisteredChannels as jest.Mock ).mockReturnValue( {
+			refetch: jest.fn(),
+		} );
+		render( <CreateNewCampaignModal onRequestClose={ () => {} } /> );
+
+		expect( screen.getByText( 'Google Ads' ) ).toBeInTheDocument();
+		expect(
+			screen.getByText(
+				'Boost your product listings with a campaign that is automatically optimized to meet your goals.'
+			)
+		).toBeInTheDocument();
+
+		// Click button to expand recommended channels section.
+		await userEvent.click(
+			screen.getByRole( 'button', {
+				name: 'Add channels for other campaign types',
+			} )
+		);
+
+		expect(
+			screen.getByText( 'Pinterest for WooCommerce' )
+		).toBeInTheDocument();
+
+		expect(
+			screen.getByText(
+				'Amazon, eBay & Walmart Integration for WooCommerce'
+			)
+		).toBeInTheDocument();
+	} );
+
+	it( 'does not render recommended channels section when there are no recommended channels', async () => {
+		( useNewCampaignTypes as jest.Mock ).mockReturnValue( {
+			data: [
+				{
+					id: 'google-ads',
+					icon: 'https://woocommerce.com/wp-content/uploads/2021/06/woo-GoogleListingsAds-jworee.png',
+					name: 'Google Ads',
+					description:
+						'Boost your product listings with a campaign that is automatically optimized to meet your goals.',
+					createUrl:
+						'https://wc1.test/wp-admin/admin.php?page=wc-admin&path=/google/dashboard&subpath=/campaigns/create',
+					channelName: 'Google Listings and Ads',
+					channelSlug: 'google-listings-and-ads',
+				},
+			],
```

Comment:
I thought about that. If I made it into an empty array, it might be a bit confusing because the line after it also uses empty array for `useRecommendedChannels`, so there would be two empty arrays for the test. I thought it would be clearer if one array is empty and one is not.

I guess your concern is about the big bunch of text hurting readability a bit. I made a refactor in [`29d67a3` (#37044)](https://github.com/woocommerce/woocommerce/pull/37044/commits/29d67a3a21d6ee5441218c0f611aef092380abab), I think it looks much better now.

Reply:
Mainly because that data is not relevant to this test objective.

> I thought about that. If I made it into an empty array, it might be a bit confusing because the line after it also uses empty array for `useRecommendedChannels`, so there would be two empty arrays for the test. I thought it would be clearer if one array is empty and one is not.

Got it. Making the dummy data share is also good. Thanks!